service/rhc_connection_validation.go

```diff
@@ -0,0 +1,29 @@
+package service
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/RedHatInsights/sources-api-go/model"
+	"github.com/RedHatInsights/sources-api-go/util"
+)
+
+// ValidateRhcConnectionRequest validates that the incoming input is valid.
+func ValidateRhcConnectionRequest(req *model.RhcConnectionCreateRequest) error {
+	if req.RhcId == "" {
+		return errors.New("the Red Hat Connector Connection's id is invalid")
+	}
+
+	sourceId, err := util.InterfaceToInt64(req.SourceId)
```

Comment:
I think this needs to be `util.InterfaceToInt64(req.SourceIdRaw)`

Reply:
Nice catch. Refactored.